Start careers infrastructure and a stab at the devops position posting. (#169)
haskellfoundation/haskellfoundation.github.io: careers/devops.markdown

---
title: DevOps Engineer
summary: The Haskell Foundation seeks a passionate DevOps professional to support GHC, related Haskell projects and tooling, and the HF itself.
postedOn: 2021-12-28
status: Open
---

# Job Details

> ## Responsibilities

Reviewer: I think this list would read better if all bullet points had the same part of speech. Verbs look like the right pick here.

* Maintain GitLab and CI infrastructure for GHC
* Improve CI

> * Expand the CI infrastructure to include base the base Haskell ecosystem

Reviewer: ```suggestion
* Design and implement an expansion of the CI infrastructure to include key components of the Haskell ecosystem
```

I think pointing out that the hire is responsible for design is important here.

Unless you're going for `base`, specifically? But I think that's already covered in today's CI setup, actually.

> * Support other Haskell libraries and tools, including HLS (Haskell Language Server)

Reviewer: How does this interact with the previous point?

> * Work with the Haskell Infrastructure Team

Reviewer: This is vague to me. Would you expect the hire to, essentially, become the functional arm of the infra team? Help out when needed? Somewhere in between? It might be good to get an idea from the infra team what they would hope from this position.

Also good would be defining the infra team, with a link to https://github.com/haskell-infra/haskell-admins

Reviewer 2: we don't imagine there will be much at first -- just over time we'd ideally integrate this person into the infra team as time becomes available and projects arise.

> * Support OS packaging of GHC and tools

Reviewer: What does this entail? Just building the binaries? Another way to ask this: how does this bullet influence how we might interview / select an applicant? Do we, for example, require the applicant to be familiar with packaging requirements for all three major OSes?

Reviewer 2: i hope not. that's something that as it becomes necessary can be learned

> * Documentation

Reviewer: of what? Haskell generally or CI specifically?

Reviewer 2: CI specifically. this isn't a tech writer slot.

> * Improvements and expansion of the GHC performance dashboard

Reviewer: To me, this bullet requires different expertise than all of the other bullets. This requires a degree of web design skills, as well as tool building. Or maybe integrating with Grafana makes this easy? Then it might be worth describing that the job is about feeding data into Grafana.

I would love it if our hire could do this, but this may be a stretch goal. Happy for other, more knowledgeable people to say that this is routine and is a straightforward expectation of a devops hire.

Reviewer 2: agreed. i think we should put "work with haskell infra team" on down in "additional responsibilities could grow to include" -- they're all things that will be figured out over time. the primary responsibilities are the first bullet points.

> ## Qualifications

Reviewer: Similarly to the section above, these should be in parallel. As these are qualifications, I personally think nouns are better here.

* Demonstrated ability to manage and simplify complexity
* Collaborate with a variety of stakeholders

> * Can juggle multiple, conflicting priorities

Reviewer: ```suggestion
* Can juggle and choose among multiple, conflicting priorities
```

> * Excellent communication skills

Reviewer: ```suggestion
* Excellent written communication skills
```

I would hope that there are also excellent oral communication skills, but our culture is strongly one of the written word.

The Haskell Foundation is looking for a DevOps engineer to take the Haskell ecosystem, primarily the Glasgow Haskell Compiler (GHC, the Haskell compiler), to the next level of stability, reliability, and performance.

This role spans many Haskell ecosystem projects to homogenize and improve the infrastructure used to build GHC and other core open source projects whose maintainers are interested, for instance HLS, Cabal, Stack, GHCup, core libraries, and more.

> The successful candidate will be in the middle of it all, collaborating with developers at the forefront of making Haskell a groundbreaking, influential, and top tier programming language. The individual will improve existing infrastructure, work on creating a shared CI and performance benchmarking system across the Haskell ecosystem, shorted CI turnaround times, and improve debugability of failures.

Reviewer: ```suggestion
The successful candidate will be in the middle of it all, collaborating with developers at the forefront of making Haskell a groundbreaking, influential, and top-tier programming language. The individual will improve existing infrastructure, work on creating a shared CI and performance benchmarking system across the Haskell ecosystem, shortened CI turnaround times, and improved ability to debug failures.
```

I feel like when I've seen "successful candidate" wording, the text is about what the candidate has done, not what the candidate will do.

> 

Reviewer: I'm left with some questions about the role. (I know the answers to some of these from other communication, but the answers should be in this posting.)

* Who would this person report to?
* Who would this person work with on a daily basis? What is the nature of the existing team?
* Where can we learn more about existing infrastructure?
* Does the applicant need to know Haskell?
* What specific technologies does the applicant need to know?
* What level of seniority are we looking for? Is there an experience requirement?
* Is there an education level requirement?
* Is the job on-site or remote?
* Are there geographical restrictions? Time zone preferences / restrictions?
* What's the HF's policy around flexibility in working conditions? What about time off? (While I don't think the job posting needs to have details here, having some idea of what to expect -- especially if the description is enticing -- may draw in more applicants.)
* What is the goal of the Haskell Foundation? What *is* the Haskell Foundation anyway? How is it funded?

Most job postings have some text around equity ("equal opportunity employer"). I'm not sure if this is legally required or just a good idea, but we should probably have such text.

Might also be good to link to the original proposal somewhere. That gives a nice dose of context and rephrases some of the job requirements (which may help readers understand better what's going on).

Reviewer 2: hiring descriptions usually don't describe the answers to 1-3. i think there should be no hard exp or education requirements but obviously those qualifications will tend to matter, but there can be balances between one and the other as well.

Also the last one really can be determined by browsing... the entire rest of the site..

Reviewer: I agree that job postings don't often contain much about 1-3, but I think clarifying these may be important here: there is currently 1 employee of the HF (and in a mostly non-technical role), and I would want an applicant to know they would be working with more than just that 1 employee (which an applicant might otherwise think). My hope is that working with several others might be seen as a good thing and encourage more applications. If it doesn't fit in the context of the rest of the text, I could see skipping this.

I'm agnostic as to any specific education or experience requirements -- I see advantages and disadvantages in every direction. But it might be good to say what the requirements are (or that we don't have any), lest an applicant erroneously think "I don't have a master's in CS, so I can't apply".

As for the last question: that was inspired by looking at several job descriptions out there for similar positions, which do tend to have a few sentences introducing the company (even when that company is very well known). I don't feel strongly (because an applicant can easily find the information), but it did seem to be part of several other postings.

Reviewer 3: > Who would this person report to?

It is pretty important to mention properly, as this is a specific context and an attractor. That the person can mostly be independent is a high attraction for professionals, DevOps'es maybe love to be even more antisocial than Devs. The "ability to work independently/autonomous" is at least good to mention. But, yes - an aware DevOps needs a ruling organ to be able to go and:
  1. report difficult problems that need money or specific specialist, or negotiate migration
  2. determine priority during  "juggle and choose among multiple, conflicting priorities" - the person needs to be mostly independent, but in overloaded situations DevOps needs to notify and establish hard choices prioritization with management,
  3. escalate the problem
  4. - important - gather the vies of the end customers & probe & use their value system.

In the original idea of "agile" is for engineering to be oriented on the value system of the customer. For DevOps - the end customer is Haskell engineers, so DevOps needs to have the freedom to discuss minor things in the chatroom and for the Haskell committee of engineers to be open to get the input questions from DevOps & give the output, as smart architect/DevOps would rely on the "terminal device" value-design system and understanding as a basis for design. Haskell DevOps needs to be a member of the group, to attend the meetings to I/O information on design and understanding.

> What specific technologies does the applicant need to know?

Is one of the most important for DevOps people. DevOps job posting without listing of technologies just reads/looks weird, as there is always happy/useful information to transmit through the enumeration of tech.

DevOps'es strongly review/judge/infer work by a stack of technologies mentioned. As DevOps has a lot of tools / DSLs - the listing has enough information to infer a lot about the work.
  * Current infrastructure listing of terms (so DevOps knows the starting point)
  * Imaginary desirable infrastructure listing to arrive to (to select people who know those technologies & can/motivated to arrive there & show that you give them technological futurism/freedom to construct infrastructure)
    * Terraform and containerization are probably worth mentioning in the listing as a desirable end point in any case.
    * Despite complexity & difficulties in the Cabal packaging - it is structurally isolated from the DevOps infrastructure. That architectural boundary needs to be revealed.
      * Probably, portion by portion Haskell DevOps infrastructure can be gradually migrated to the Nix stack.
      * From experience can tell that the person that knows Haskell - understands/knows/easily learns Nix better - and would learn, know & write a better Nix code than a person which learned Nix only. So the person that both knows Haskell & Nix somewhat - (with the caveat of packaging design boundary, otherwise) has mutual synergy with Haskellers & becomes a better Nix'er.
      * Usage of NixOS in the long term would save DevOps maintenance efforts & infrastructure.
      * Nix establishes a strong guarantee/requirement that almost all infrastructure would be saved as an infrastructure as a code. Person does not need to be controlled - Nix requires DevOps to be done as infrastructure as a Nix code.
      * As a result, in any situations infrastructure as a code would be easy to hand over.
      * Would be easy to add people into the Ops team, easy to extend it with "scratch own itch" volunteer flow.
      * It would be easy to inspect projects and infrastructure & would be easy for the community to chime in with Pull Requests. So person would not be totalitarian-like alone with infrastructure, it would be a community process.
      * NixOS/Nixpkgs community & expertise is vast.
      * A lot of people want to work in Nix DevOps
      * NixOS/Nixpkgs DevOps talent pool is not tapped into by business. It is a rich and motivated talent pool.
      * The NixOS/Nixpkgs DevOps talent pool has a community collaboration value system that aligns with what Haskell Foundation nature.
      * So in NixOS/Nixpkgs community - a great number of experienced DevOps'es can be found & eager to work in Haskell Foundation.
      * The positive community development feedback loop is strong.
      
Overall the current stack needs to be described, and the second listing is the fictional end stack as an example.

Reviewer 3: If Nix would be considered - we togather happen to know/be connected to/have a network of a lot of Nix DevOps mastodonts. It would be immensely useful to send them information directly.

Reviewer 4: This person will (to my knowledge) report to @bgamari.

This is a very versatile role.  The tech stack is a mixture of nix (some PS1 scripts on windows) and gitlab. One of the key points to consider is that we run on a lot of donated hardware, so access is not always trivial (this depends a lot of building up trust) and might end up needing to be communicated with third party providers. If someone brings in some Haiku builder, I'm fairly sure we can't use nix on that.

Core skills to me are
- excellent analytical
- excellent social
- good ops experience
- good haskell knowledge (you don't need expert haskell knowledge, but you need to at least be able to understand why some thing might fail. We have haskell engineers to fix bugs, the CI is *not* written in haskell).
- gitlab experience preferably 
- data collection, visualisation and interpretation experience would be good
- optional macOS experience (we have lots of people comfortable with linux, but debugging macOS issues not so many)

Now the tech stack is not completely fixed, but we do build on linux, macOS, windows, bsd, and various architectures. Some can be managed by nix, some can't.

This is the core CI part. Issues that come up are mostly: some builder starts acting up, what's wrong, why does that builder behave strange?

Collect performance metrics of builds, and figure out if we can improve performance.

Reviewer: > This person will (to my knowledge) report to @bgamari.

This surprises me. For me, the "reports to" relationship implies that e.g. @bgamari would have the authority to fire the devops person, likely in consultation with others. It also implies that when the hire is ready to ask for a raise, they would ask @bgamari -- or that @bgamari would have a definitive say on the matter. It would be unusual (to me) to have that relationship fall outside the HF. I'm not saying this is problematic, per se, but it does strike me as unusual and something that should be explicitly agreed to on all sides.

Reviewer 3: https://github.com/haskellfoundation/haskellfoundation.github.io/pull/169#discussion_r782678617 sounds to me, as I pointed out "in any case", because indeed Nix can not be/sometimes is not effective to be used - so the use of Terraform, Docker & Ansible is required.

Reviewer 3: If a person reports to CTO, it does not mean that CTO is who decides to fire/hire a person or decides on monetary spending. The money/fire/hire questions are generally fit for collective/hybrid decision structure, while it is also nearly mostly impossible to report/have as a direct order from the collective, because for that to function - it implies a requirement of quorum from a collective to agree on decisions. While DevOps frequently has a high level of requirement for stochastic reporting & negotiation. Technical bug, hardware, DDoS, or service - DevOps may be required to do sudden work with hard choices & the chief negotiation then is needed, & nobody would collect a group of people fast enough for that, so having 1 person to report to is needed.

Reporting/decision making from a collective requires an overhead from the collective & the other part - is that in the collective the responsibility gets split while on requests/dialog through time certain parts of a collective can be unaware of prior decisions & give contrary decisions/responses.

The etymology of "manager" alludes to why that point in structure exists & why we keep semantically meaning of managing people.

It is mostly desirable for the 1 formal I/O entry point to exist/be determined, and for people to build relationships and understanding between each other & establish the style of the work process.

`design/todo` flow is a different process from hiring/money decision-making. The hiring/money questions are generally ones that escalate way beyond the worker-manager, as, I believe, HF already has a monetary organ for example.

Reviewer 5: Indeed I would hope that there would be a contact within HF who would be responsible for "HR" relevant interactions. However, as I've stated earlier I would be happy to help set tasks and priorities.

Reviewer: This is a key bit:

> as I've stated earlier I would be happy to help set tasks and priorities

Thank you for doing this! However, there is (in my mind) a biggish difference between "help set tasks and priorities" and "set tasks and priorities". The former calls to mind regular checkins with Andrew, who assigns tasks, sometimes informed by Ben's technical details (say). The latter says that Ben chooses the tasks and assigns them. The current text reads like the latter, but Ben's response says the former.

I could be splitting hairs here. (Actually, I *am* splitting hairs. It's just that I think this is an important hair to split!) It just seems really easy to me to have miscommunication and mistaken expectations here, and a mismatch here would have a very long shadow.

Reviewer 3: It is indeed true.

The wording of the application posting conveys a lot to the job seeker, as that is all information person often has about the work & climate in the company. Indeed, the simple word "help" implies a tremendous difference in reading & understanding, as it moves from the rigid two-way architectural dependency between worker & manager, into a lax one. From totalitarian into actor model.

The analog of difference in law or standards would be the `may` word.

---
title: DevOps Engineer
summary: The Haskell Foundation (HF) seeks a passionate DevOps professional to support GHC, related Haskell projects and tooling, and the HF itself.
postedOn: 2022-01-12
status: Open
---

# Description

The [Haskell Foundation](https://haskell.foundation) is [looking for a DevOps engineer](https://github.com/haskellfoundation/tech-proposals/blob/main/proposals/accepted/003-ghc-ops.md) to take the Haskell ecosystem, primarily the [Glasgow Haskell Compiler](https://gitlab.haskell.org/ghc/ghc) (GHC, the Haskell compiler), to the next level of stability, reliability, and performance.

> This role spans many Haskell ecosystem projects to homogenize and improve the infrastructure used to build GHC and other core open source projects whose maintainers are interested, for instance HLS, Cabal, Stack, GHCup, core libraries, and more.

Reviewer 6: Yes -- I could barely parse it, let alone understand it.

How about this as an intro

--------------------

The HF is looking for a DevOps engineer to take the Haskell ecosystem to the next level of stability, reliability, and performance.

The principal initial focus is to solidify and develop the CI infrastructure for GHC.  But once that is done, we anticipate offering the HF's CI infrastructure support to other core open source projects, should they wish to take adfvantage of it.  Examples of such projects inclue Cabal, Stack, GHCup, Haskell Language Server and core libraries.  

The successful candidate will have exceptional *technical skills* in devops infrastructure (build systems, cloud services, ... [not sure what esles to say].   Becuase the role involves deep collaboration with the developers of Haskell tools and librairies, the successful candidate will also have very strong *communication skills*.

The successful candidate will be in the middle of it all, collaborating with developers at the forefront of making Haskell a groundbreaking, influential, top-tier programming language. The individual will improve existing infrastructure, work on creating a shared [CI](https://gitlab.haskell.org/ghc/ghc/-/tree/master/.gitlab) and performance benchmarking system across the Haskell ecosystem, shortened CI turnaround times, and improved ability to debug failures.

# Responsibilities

> * Maintain CI (continuous integration) infrastructure for GHC (the Glasgow Haskell Compiler, the leading compiler for the Haskell language), as hosted on GHC's GitLab instance.

Reviewer 6: Maintain and develop

* Work with the GHC team to improve CI efficiency, reliability, and resource utilization.

> * Design and implement expanding the CI infrastructure to include key components of the Haskell ecosystem.

Reviewer 6: I'm not sure what this bullet really means.   Perhaps it means

* Design and implement shared CI infrastructure that can be offered to the developers of other key Haskell tools and libraries, in consultation with those developers.

> * Design and implement expanding the CI infrastructure to include key components of the Haskell ecosystem.

Reviewer 5: There is a bit of a difference between what Simon writes and the original text: the latter emphasizes that we are extending the scope of an existing system. I think this is probably good message to reinforce.

> * Support other Haskell libraries and tools, including HLS (Haskell Language Server).

Reviewer 6: Is that different to the previous item?

Reviewer 5: I'm also a bit unclear on what this means.

> * Support OS packaging of GHC and tools.

Reviewer 6: Say a bit more?   We are talking about (a) CI and (b) OS packaging, right?  For both GHC and other tools.

Reviewer 5: ```suggestion
* Support operating-system-dependent packaging of GHC and related tools.
```

> * Documentation for the CI infrastructure.

Reviewer 6: Or "CI and OS packaging infrastructure"?

## Likely future responsibilities

* Work with the [Haskell Infrastructure Team](https://github.com/haskell-infra/haskell-admins/).
* Improvements and expansion of the GHC performance dashboard.

# Qualifications

* Demonstrated ability to manage and simplify complexity
* Collaborate with a variety of stakeholders
* Ability to choose between multiple, conflicting priorities
* Excellent English written communication skills

# Current Technology Stack

* Most GHC services are hosted by a single machine running NixOS. This includes our PostreSQL database, GitLab and Grafana instances, our Docker registry, as well as various project-specific services. All of these services are provisioned via their associated Nix modules.
* Our CI infrastructure builds upon GitLab Pipelines, using Docker for build isolation and reproducibility on Linux. A qualified applicant should be comfortable with both technologies.

> * We maintain a fleet of heterogeneous CI runners, covering Linux (AArch64, x86-64), Windows (x86-64), and Darwin (AArch64).

Reviewer 5: I suppose we could also include x86-64/Darwin here as we do have one such runner.

> * We maintain a fleet of heterogeneous CI runners, covering Linux (AArch64, x86-64), Windows (x86-64), and Darwin (AArch64).

Reviewer 4: We do build x86_64 on aarch64-darwin machines. This is faster than building it on what ever x86_64-darwin machines apple officially economically offers. It's somewhat ridiculous that it's faster to run through the x86_64 emulation layer on an aarch64 apple machine than on available x86_64 hardware. Again, we *do* build for x86_64-darwin and from the tools it looks like they are on x86_64-darwin, it's just a full architecture translation beneath.

# Bonus Points

* Some famliarity with Haskell packaging.
* Familiarity with platform-dependent software packaging (e.g. notarisation and code signing on Darwin).
* Some systems programming experience.

# Misc.

This is a 100% remote position, reporting to the Executive Director of the Haskell Foundation, but taking technical direction from the GHC team (primarily Ben Gamari). Primary interactions will include people in US and European time zones, but can include coordination with people across the world at times.

> Haskell knowledge in not required.

Reviewer: ```suggestion
Haskell knowledge is not required.
```

> Budgeted maximum total compensation for this position is $$124k USD. Actual terms depend on experience, expertise, geographical location, local employment laws, etc.

Reviewer 7: Should `$$124k` be `$124k`?

Author: That's to handle Hakyll macros, it becomes $.

> Budgeted maximum total compensation for this position is $$124k USD. Actual terms depend on experience, expertise, geographical location, local employment laws, etc.

Reviewer 3: ![Screenshot-2022-01-13-19:50:15](https://user-images.githubusercontent.com/20933385/149382703-120fff2f-226a-4e07-942a-b5e829a7b647.png)

This is kinda offensive.

Of course, as in Google Summer of Code put it - they give equal opportunity to people - the same students for the same work done in different countries would receive ~1K in Somalia, & 5K in Switzerland. Indeed that is equal encouragement & opportunity. Since smart people can't exist/survive in Somalia, since if they educated themselves smart enough to work in GHC - as my friends told - people with higher education, the kind of polite smart people - are considered by military regimes a too valuable commodity to give up during the regime change & overall get considered a hazard to local military rulers. So friends told that on return to homeland they would be killed since they received higher education, a father of one of them, electrician, manages to not get killed, because the local rulers need electricity.
I know that the argument of the multiplication of a person's life on "The Economic Power Index of the country" works great on paper. Indeed people who managed to keep high morals & stay good people in Somalia, do not need money or help or encouragement.

---

Depending on my GPS location - I get drastically different salaries & work offers (in fact living here I get no offers at all, only automatic replies). I would uncover a secret, in Ukraine median salary is ~4323.6$. It is a year. Because, depending on the motivation, I'm not fancy & I can't get work (except rare invitations to do cryptocurrencies, which I begin to consider), I can survive here on those 360$ a month & would be ready to go damping that low, because I have high morals & disregard money reasoning. In addition to what is requested in the requirements I would also have the knowledge of Haskell tooling, and happen to Haskell on some level, so I guess that can be used as a starting point & people while live in Ukraine probably deserve it. But wait, there are countries where is an even cheaper workforce, so even those 360$/m can be saved depending on the geographic locations. There is a lot of economies that are worse than in Ukraine, where people live even in more peculiar situations.

---

I guess these are true facts nevertheless. If work is marked fully remote, at least do not advertise to a person that how much a person would earn for the same fully remote work depends on GPS location, because you would divert good specialists in poor countries & encourage people in rich countries to apply because of money. That is an organization's internal considerations and level of experience & bargaining the salary exists for it. The normal salary bargaining happens in the normal conditions. It is Ok to ask for "raw expectations" during interviews & establish the salary. Also probably it is better to not tell the maximum sum - because ... there seem to be no good logical paths of good outcomes from that. If the person would cost all $124K - you probably know that person & if they are interested - they would probably 1. be noticed on application 2. and would ask "how much HF got" in the chatroom or in the interview, if they need that much money.

In any situation: "The position was evaluated $124k, but I was chosen as the best candidate & highered at 48k" - that fact stays in the back of the head of the person during work, demotivating a person or can start processes of internal reasoning, in a long term that can play a role between during hard moments of high complexity tasts, high demand in work & overtimes - the person being motivated or not, if the person would know that position was $124k, but they do work for X$ - it can demotivate the person.

I suggest only the gradual Newtonian method of ballparking around the normal distribution deviation. People in capitalism seem to forget how to bargain. I still remember why the tradition of bargain existed, in childhood, it told me that character & people communication & people values worth more than prices, which frequently have a high degree of arbitrary.

I love honesty & mainly prefer it. But I am not monadic to honesty. Sometimes it is easier to pretend you do not know, or ascent to the level of to not care about what other people do/how they live, than to live, explicitly see &, live through it & feel it every day while you can't change it really for you, only you for future generations or society as a whole can change.

So, as a poor person from a poor country - I would ask to remove the mentioning that salary depends on "geographical location". And the upper bound of salary always seemed illogical to me, if you want to select people - set the minimal salary - that would show to applicants how much serious person needs to be to the requirements, the people that want too much to their level - is in itself a redflag to the quality of the person. The former one (country) - both parties know & it is a Polichinelle secret, the second one (max boundary) is illogical to mention, people anyway should not apply for profit reasons into nonprofit organizations, I think Haskell Foundation can find a DevOps for a reasonable cost without mentioning the upper bound.

> Budgeted maximum total compensation for this position is $$124k USD. Actual terms depend on experience, expertise, geographical location, local employment laws, etc.

Reviewer: Thanks @Anton-Latukha. There's a lot there, but perhaps we can extract out this last piece: does it make more sense to mention a lower bound than an upper bound? I wrote a bit about how I was worried about stating an upper bound, but I then deleted it, as I wasn't sure -- and I think it's helpful to applicants to give an idea of what we're thinking about. We can give that idea via a lower bound instead. I don't know what's best here -- this is far from my area of expertise and experience -- but I think it's an interesting idea to consider.

> Budgeted maximum total compensation for this position is $$124k USD. Actual terms depend on experience, expertise, geographical location, local employment laws, etc.

Reviewer 3: I used lower bound as an example, I not imply as it should be used.

I think that overall simply the "market competitive" salary indication should be used. But because standard wordings get plundered - the new wording needs to be constructed to imply the same.

Like, "we would be able to provide a normal market price for according experience in work".

> Budgeted maximum total compensation for this position is $$124k USD. Actual terms depend on experience, expertise, geographical location, local employment laws, etc.

Reviewer 3: During job interviews simply reinstating that the position funding abilities are as in normal DevOps market - already enough.

Almost all people who would apply & you would review would land into 3k-9k range anyway, in that condition, the precise price can be negotiated in the end. I think most people who recently had earned DevOps salaries can wait a week or two for HF to ask a bunch of people on salary & compare them.

Also, the ["best bride marriage problem solution"](https://en.wikipedia.org/wiki/Secretary_problem) in optimization theory is good to know (evaluate people for 1/e interval of hiring time range & then start searching for the next best one), also the real-life better than "best bride", the people actually can be called back in time.

> Budgeted maximum total compensation for this position is $$124k USD. Actual terms depend on experience, expertise, geographical location, local employment laws, etc.

Reviewer 3: So, that solution needs "the time range to hire".

Dice analytics in the salary report, report 53 days to hire a DevOps
![Screenshot-2022-01-14-23:02:24](https://user-images.githubusercontent.com/20933385/149584815-988776e0-51ad-4b88-b6d8-c8b796eaad77.png)
(source: Dice 2021 Tech Salary Report)
